### DISCUSSION TITLE
Support for Morph To & Morph Many relationships

### DISCUSSION THREAD

**Discussion by duncanmcclean** [statamic-rad-pack/runway#245] [Feature Requests]
It would be great if Runway could support `MorphTo` relationships. 

For example, I have this `commentable` relationship which can either be a `CommentableItem` or a `Comment` provided by the Spatie Comments package.

This relationship essentially represents what the comment is commenting on.

<img width="393" alt="image" src="https://user-images.githubusercontent.com/19637309/230710594-8f47b795-a019-4e42-a9d1-afa85bc9f070.png">

Ideally, the 'Morph To' fieldtype would work similarly to the 'Belongs To' fieldtype. However, there's a few things that need to be considered:

* How do we let the user pick from *all* models? (*all* because I think there's a config option for which models can be used on a morph relationship)
* Should we show which resource the attached model is (on the fieldtype - I think the Entries fieldtype does this)

**Comment by mefenlon** [comment]
I am using Filament as a backend for a project and runway for the frontend. It may be worth having a look at how Filament is dealing with the relationship field.
https://filamentphp.com/docs/3.x/forms/fields/select#handling-morphto-relationships

**Comment by mefenlon** [comment]
> - How do we let the user pick from all models? (all because I think there's a config option for which models can be used on a morph relationship)


I see three ways of accomplishing this:

1. Define an array setting in the config. Something like `'morphs' => ['articles', 'pages']`
2. Configure in the blueprint or in the fieldtype interface using resource names
```
-
    handle: comments
    field:
        type: morph_to
        resource: comment
        mode: default
        morphs:
            - 'articles'
            - 'pages'
```
3. Loop through all resources and see if the inverse relationship exists model has comments() defined. I think this will need to be done in the above cases as well to make sure the relationship is possible, so maybe default to this method and then be able to limit by ont of the above methods.
```
//If defined in model, list in select field
    public function comments(): MorphToMany
    {
        return $this->morphToMany(Comment::class, 'commentable');
    }
```

> - Should we show which resource the attached model is (on the fieldtype - I think the Entries fieldtype does this)


1. Using `'title_field'` that is already defined in the config followed by the resource or class name 
_Title of an article [article]
Title of a page [page]_
2. Defining an Accessor in the model
```
    public function morphTitle(): Attribute
    {
        return Attribute::make(
            get: fn () => $this->title . $this->date,
        );
    }
```
